webui/react/src/stores/workspaces.tsx

```diff
@@ -1,194 +1,135 @@
+        Loadable.map(loadable, (workspaces) => {
+          return workspaces.map((workspace) => {
+            return workspace.id === id ? { ...workspace, pinned: false } : workspace;
+          });
+        }),
+      ),
+    );
+  }
 
-  return useCallback(
-    (id: number, updater: (arg0: Workspace) => Workspace): void => {
-      workspaces.update((ldbWorkspace) => {
-        return Loadable.map(ldbWorkspace, (ws) =>
-          ws.map((old) => (old.id === id ? updater(old) : old)),
-        );
+  public createWorkspace(params: V1PostWorkspaceRequest): Promise<Workspace> {
+    return createWorkspace(params).then((workspace) => {
+      this.#loadableWorkspaces.update((loadable) => {
+        return Loadable.map(loadable, (workspaces) => [...workspaces, workspace]);
       });
-    },
-    [workspaces],
-  );
-};
-
-export const useCreateWorkspace = (): ((arg0: V1PostWorkspaceRequest) => Promise<Workspace>) => {
-  const context = useContext(WorkspacesContext);
-  if (context === null) {
-    throw new Error('Attempted to use useCreateWorkspace outside of Workspace Context');
+      return workspace;
+    });
   }
 
-  const { workspaces } = context;
-
-  return useCallback(
-    async (params: V1PostWorkspaceRequest): Promise<Workspace> => {
-      const createdWs = await createWorkspace(params);
-      workspaces.update((ldbWorkspaces) => {
-        return Loadable.map(ldbWorkspaces, (ws: Workspace[]) => [...ws, createdWs]);
-      });
+  public deleteWorkspace(id: number): Promise<void> {
+    return deleteWorkspace({ id }).then(() =>
+      this.#loadableWorkspaces.update((loadable) =>
+        Loadable.map(loadable, (workspaces) => {
+          return workspaces.filter((workspace) => workspace.id !== id);
+        }),
+      ),
+    );
+  }
 
-      return createdWs;
-    },
-    [workspaces],
-  );
-};
+  public fetch(settings = {} as GetWorkspacesParams, signal?: AbortSignal): () => void {
+    const canceler = new AbortController();
 
-// On logout, clear old workspace records.
-export const useResetWorkspaces = (): (() => void) => {
-  const context = useContext(WorkspacesContext);
+    getWorkspaces(settings, { signal: signal ?? canceler.signal })
+      .then((response) => this.#loadableWorkspaces.set(Loaded(response.workspaces)))
+      .catch(handleError);
 
-  if (context === null) {
-    throw new Error('Attempted to use useResetWorkspaces outside of Workspace Context');
+    return () => canceler.abort();
   }
 
-  const { workspaces } = context;
-
-  return useCallback((): void => {
-    workspaces.set(NotLoaded);
-  }, [workspaces]);
-};
-
-export const useDeleteWorkspace = (): ((id: number) => Promise<void>) => {
-  const context = useContext(WorkspacesContext);
+  public reset() {
+    this.#loadableWorkspaces.set(NotLoaded);
+  }
 
-  if (context === null) {
-    throw new Error('Attempted to use useDeleteWorkspace outside of Workspace Context');
+  protected async poll(settings: GetWorkspacesParams = {}) {
```

Comment:
Workspaces are only ever updated by the frontend, we don't need to poll for them.

Reply:
the current state of `Navigation` and `ModelRegistry` components both poll for workspaces, so it was more about not changing that behavior.